pkg/live/apply-crd-task.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package live
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/resource"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util"
+	"sigs.k8s.io/cli-utils/pkg/apply/taskrunner"
+)
+
+// ApplyCRDTask encapsulates information necessary to apply a
+// Custom Resource Definition (CRD) as a task within a task queue.
+// Implements the Task interface.
+type ApplyCRDTask struct {
```

Comment:
Do we need a new Task type for this? I would think that just using the existing `ApplyTask` should work.

Reply:
I tried that, but it assumes an inventory object (for InventoryPolicy) which is not involved in this operation.